File: src/app_logic/generate_password.py
from random import choice
from string import ascii_lowercase, ascii_uppercase, digits, punctuation


class GeneratePassword:
    def __init__(self):

        self.numbers = digits
        self.lowercase = ascii_lowercase
        self.uppercase = ascii_uppercase
        self.symbols = punctuation

        self.total_password = (
            self.lowercase + self.uppercase + self.symbols + self.numbers
        )

    def generate(self, password_length, page):
        self.password = "".join(
            choice(self.total_password) for i in range(password_length)
        )

        page.set_clipboard(self.password)

        return self.password

File: internal/password/generate_password.go
package password

import (
	"math/rand"
	"strings"
	"time"

	"github.com/RaulCatalinas/PasswordGenerator/internal/constants"
)

type passwordGenerator struct{}

func NewPasswordGenerator() *passwordGenerator {
	return &passwordGenerator{}
}

func (pg *passwordGenerator) GeneratePassword(passwordLength int) string {
	totalChars := constants.LOWERCASE + constants.UPPERCASE + constants.SYMBOLS + constants.NUMBERS

	// Initialize random seed
	rand.New(rand.NewSource(time.Now().UnixNano()))

	var password strings.Builder

	password.Grow(passwordLength)

	for i := 0; i < passwordLength; i++ {
		randomIndex := rand.Intn(len(totalChars))
		password.WriteByte(totalChars[randomIndex])
	}

	return password.String()
}
